File: core/src/org/rexellentgames/dungeon/entity/item/accessory/equipable/ChallengeRune.java
package org.rexellentgames.dungeon.entity.item.accessory.equipable;

import org.rexellentgames.dungeon.assets.Locale;
import org.rexellentgames.dungeon.entity.creature.mob.Mob;

public class ChallengeRune extends Equipable {
	{
		name = Locale.get("challenge_rune");
		description = Locale.get("challenge_rune_desc");
		sprite = "item (scroll D)";
	}

	@Override
	public void onEquip() {
		super.onEquip();
		Mob.challenge = true;

		for (Mob mob : Mob.all) {
			mob.generatePrefix();
		}

		this.owner.modifyDefense(2);
	}

	@Override
	public void onUnequip() {
		super.onUnequip();
		Mob.challenge = false;
		this.owner.modifyDefense(-2);
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/ChallengeRune.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale
import org.rexcellentgames.burningknight.entity.creature.mob.Mob

class ChallengeRune : Equippable() {
	init {
		name = Locale.get("challenge_rune")
		description = Locale.get("challenge_rune_desc")
		sprite = "item-scroll_d"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		Mob.challenge = true

		for (mob in Mob.all) {
			mob.generatePrefix()
		}

		this.owner.modifyDefense(2)
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		Mob.challenge = false
		this.owner.modifyDefense(-2)
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}

	override fun canBeDegraded(): Boolean {
		return false
	}
}